### PR TITLE
Remove ksLookup features

### LINKED ISSUE #3083: remove ksLookup features not related to looking up keys
Elektra used to have ksLookup features that were simply iterating over all keys and checking values. This functionality is not very useful.

Thus following functions should be removed:

- [ ] ksLookupByString (which is also buggy, see a9e2aab7cc933995e99b48c0fa6c4deb75e11755)
- [ ] ksLookupByBinary
- [ ] elektraLookupLinearSearch
- [ ] KDB_O_NODIR and so on, are already marked to be removed, until
 - [ ] KDB_O_NOALL

### PR DESCRIPTION
**Pull request by PhilippGackstatter**
fixes #3083 

## Basics

These points need to be fulfilled for every PR:

- [x] Short descriptions of your changes are in the release notes
      (added as entry in `doc/news/_preparation_next_release.md` which
      contains `_(my name)_`)
      **Please always add something to the release notes.**
- [x] Details of what you changed are in commit messages
      (first line should have `module: short statement` syntax)
- [ ] References to issues, e.g. `close #X`, are in the commit messages.
- [ ] The buildservers are happy.
- [x] The PR is rebased with current master.

If you have any troubles fulfilling these criteria, please write
about the trouble as comment in the PR. We will help you.
But we cannot accept PRs that do not fulfill the basics.

## Checklist

Check relevant points but **please do not remove entries**.
For docu fixes, spell checking, and similar none of these points below
need to be checked.

- [ ] I added unit tests for my code
- [ ] I fully described what my PR does in the documentation
      (not in the PR description)
- [ ] I fixed all affected documentation
- [x] I added code comments, logging, and assertions as appropriate (see [Coding Guidelines](https://master.libelektra.org/doc/CODING.md))
- [ ] I updated all meta data (e.g. README.md of plugins and [METADATA.ini](https://master.libelektra.org/doc/METADATA.ini))
- [ ] I mentioned every code not directly written by me in [THIRD-PARTY-LICENSES](doc/THIRD-PARTY-LICENSES)

## Review

Reviewers will usually check the following:

- [ ] Documentation is introductory, concise, good to read and describes everything what the PR does
- [ ] Examples are well chosen and understandable
- [ ] Code is conforming to [our Coding Guidelines](https://master.libelektra.org/doc/CODING.md)
- [ ] APIs are conforming to [our Design Guidelines](https://master.libelektra.org/doc/DESIGN.md)
- [ ] Code is consistent to [our Design Decisions](https://master.libelektra.org/doc/decisions)

## Labels

If you are already Elektra developer:

- Add the "work in progress" label if you do not want the PR to be reviewed yet.
- Add the "ready to merge" label **if the basics are fulfilled** and you also
  say that everything is ready to be merged.

### CODE REVIEW

**Comment by markus2330** [comment]
Thank you for working on this!

**Comment by PhilippGackstatter** [comment]
jenkins build libelektra please

**Comment by PhilippGackstatter** [comment]
jenkins build libelektra please

**Comment by PhilippGackstatter** [comment]
jenkins build libelektra please

File: src/libs/elektra/keyset.c [hunk 1]
```diff
@@ -533,51 +533,6 @@ static int keyCompareByName (const void * p1, const void * p2)
...
-				ret = 1;
-			}
-		}
-	}
-	return ret;
-}
-
 /**
  * @brief Compare only the owner of two keys (not the name)
  *
```
Comment (markus2330) [inline src/libs/elektra/keyset.c]: keyCompareByOwner and keyCompareByNameOwner also is not needed anymore (but maybe @kodebach already removed that?)

Comment (kodebach) [reply src/libs/elektra/keyset.c]: No, I have not removed them.

Comment (markus2330) [reply src/libs/elektra/keyset.c]: I added them to be removed in #3124

**Comment by PhilippGackstatter** [comment]
> Looks good, is this ready to merge?

Yes ready to merge.

File: tests/abi/testabi_ks.c [hunk 2]
```diff
@@ -1033,15 +1033,15 @@ static void test_ksLookup (void)
 		ksUnsort (ks);
 		for (j = 0; j < 23; j++)
 			succeed_if (ksLookup (ks, k[j], 0) == k[j], "did not find key");
-		succeed_if (ksLookup (ks, k[23], KDB_O_NOCASE) == k[5], "did not find key");
+		// succeed_if (ksLookup (ks, k[23], KDB_O_NOCASE) == k[5], "did not find key");
```
Comment (markus2330) [inline tests/abi/testabi_ks.c]: you can completely remove this here

**Comment by markus2330** [comment]
Thank you, leftovers please in a new PR.

**Pull request merged**